internal/controllers/machinedeployment/machinedeployment_status_test.go

```diff
@@ -0,0 +1,1185 @@
+		})
+	}
+}
+
+func Test_setDeletingCondition(t *testing.T) {
+	tests := []struct {
+		name                                         string
+		machineDeployment                            *clusterv1.MachineDeployment
+		machineSets                                  []*clusterv1.MachineSet
+		getAndAdoptMachineSetsForDeploymentSucceeded bool
+		machines                                     []*clusterv1.Machine
+		getMachinesSucceeded                         bool
+		expectCondition                              metav1.Condition
+	}{
+		{
+			name:              "get machine sets failed",
+			machineDeployment: &clusterv1.MachineDeployment{},
+			machineSets:       nil,
+			getAndAdoptMachineSetsForDeploymentSucceeded: false,
+			machines:             []*clusterv1.Machine{},
+			getMachinesSucceeded: true,
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineDeploymentDeletingV1Beta2Condition,
+				Status:  metav1.ConditionUnknown,
+				Reason:  clusterv1.MachineDeploymentDeletingInternalErrorV1Beta2Reason,
+				Message: "Please check controller logs for errors",
+			},
+		},
+		{
+			name:              "get machines failed",
+			machineDeployment: &clusterv1.MachineDeployment{},
+			machineSets:       []*clusterv1.MachineSet{},
+			getAndAdoptMachineSetsForDeploymentSucceeded: true,
+			machines:             nil,
+			getMachinesSucceeded: false,
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineDeploymentDeletingV1Beta2Condition,
+				Status:  metav1.ConditionUnknown,
+				Reason:  clusterv1.MachineDeploymentDeletingInternalErrorV1Beta2Reason,
+				Message: "Please check controller logs for errors",
+			},
+		},
+		{
+			name:              "not deleting",
+			machineDeployment: &clusterv1.MachineDeployment{},
+			machineSets:       []*clusterv1.MachineSet{},
+			getAndAdoptMachineSetsForDeploymentSucceeded: true,
+			machines:             []*clusterv1.Machine{},
+			getMachinesSucceeded: true,
+			expectCondition: metav1.Condition{
+				Type:   clusterv1.MachineDeploymentDeletingV1Beta2Condition,
+				Status: metav1.ConditionFalse,
+				Reason: clusterv1.MachineDeploymentDeletingDeletionTimestampNotSetV1Beta2Reason,
+			},
+		},
+		{
+			name:              "Deleting with still some machine",
+			machineDeployment: &clusterv1.MachineDeployment{ObjectMeta: metav1.ObjectMeta{DeletionTimestamp: &metav1.Time{Time: time.Now()}}},
+			machineSets: []*clusterv1.MachineSet{
+				fakeMachineSet("ms1", withStatusReplicas(1)),
+			},
+			getAndAdoptMachineSetsForDeploymentSucceeded: true,
+			machines: []*clusterv1.Machine{
+				fakeMachine("m1"),
+			},
+			getMachinesSucceeded: true,
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineDeploymentDeletingV1Beta2Condition,
+				Status:  metav1.ConditionTrue,
+				Reason:  clusterv1.MachineDeploymentDeletingDeletionTimestampSetV1Beta2Reason,
+				Message: "Deleting 1 Machines",
```

Comment:
nit: Would be nice to differentiate "1 Machine" vs "2 Machines"  (same for MS)

Reply:
done; MS are not implemented yet